src/main/java/com/amazon/dlic/util/Roles.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.dlic.util;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class Roles {
+    private static final Logger log = LogManager.getLogger(Roles.class);
+
+    public static String[] split(Object rolesObject) {
+        if (rolesObject == null) {
+            return new String[0];
+        } else if (rolesObject instanceof Collection) {
+            List<String> roles = new ArrayList<>();
+            for (Object object : (Collection<?>) rolesObject) {
+                if (object instanceof Collection) {
+                    for (Object subObject : (Collection<?>) object) {
+                        roles.addAll(Arrays.asList(splitString(String.valueOf(subObject))));
+                    }
+                } else {
+                    if (!(object instanceof String)) {
+                        // We expect a String or Collection. If we find something else, convert to
```

Comment:
can you add a comment here what type of data will be present here ?

Reply:
when given path of JSON object instead of array, it will be` java.util.LinkedHashMap`. I will add a comment to specify the cases.